text/0000-simplify-ros-start-trajectory.md

```diff
@@ -0,0 +1,67 @@
+# Simplify start_trajectory.srv
+
+## Summary
+[summary]: #summary
+
+This RFC summarizes a few observed problems _of_ and suggestions for improvement _for_ the `/start_trajectory` service of Cartographer ROS.
+
+## Motivation
+[motivation]: #motivation
+
+Starting a trajectory on demand, optionally with a start pose, is a very common task in robot localization.
+However, the [`/start_trajectory`](https://github.com/googlecartographer/cartographer_ros/blob/3ca30fc90458152cc9a2c52aaa556e0ab09d0871/cartographer_ros_msgs/srv/StartTrajectory.srv) service is very hard to use for service clients. 
+
+Recent discussions related to this:
+
+* https://github.com/googlecartographer/cartographer_ros/issues/1193
+* https://github.com/googlecartographer/cartographer_ros/issues/1212
+* https://github.com/googlecartographer/rfcs/pull/17#issuecomment-468639993
+
+It requires a binary string of the trajectory options proto (see [here](https://github.com/googlecartographer/cartographer_ros/blob/61dd57bd94/cartographer_ros_msgs/msg/TrajectoryOptions.msg)), which is impossible to build by client code that should only know the message contract from `cartographer_ros_msgs`.
+
+The `start_trajectory_main.cc` executable exists only to make this service usable.
+It hides the implementation details away from the user/client with a proper interface that has only 3 simple options.
+
+## Approach
+[approach]: #approach
+
+Change the service definition so that clients only need to depend on message packages (`cartographer_ros_msgs`, `geometry_msgs`, `std_msgs`), e.g.:
+
+```
+string configuration_directory
+string configuration_basename
```

Comment:
As mentioned by @ojura in the open house meeting, this requires that the configuration lies on the host filesystem of the Cartographer node. Binary proto allowed to transfer options from a client in the network.

We agreed that we will drop this possibility for the sake of simplicity, which will benefit more users. Also, client/server scenarios are better covered by Cartographer gRPC anyway.